lib/fork.js

```diff
@@ -169,18 +169,23 @@ cluster.fork = function(env) {
       });
     }
   }
-  
+
+  function ondisconnect() {
+    var index = workers.indexOf(obj);
+    if (index >= 0) {
```

Comment:
@fent Hmm, I did not look through all the codes but presume that it works like: `online` `workers.push`, `disconnect` `workers.splice`, `online` again (for cluster module in future ;) `workers.push` again. However, looks like `workers.push` was not written in `online`...
I'll change `onexit` only then and make a new pr.